### PR TITLE
Fix obj address in syscall_storage_obj_create()

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
This fixes "xtest 7584"

Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

File: core/tee/tee_svc_storage.c [hunk 1]
```diff
@@ -587,7 +587,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 
 	if (attr != TEE_HANDLE_NULL) {
 		/* init attributes  if provided */
-		res = tee_obj_get(sess->ctx, attr, &attr_o);
+		void *ref = tee_svc_uref_to_kaddr(attr);
+
+		res = tee_obj_get(sess->ctx, (uint32_t)(uintptr_t)ref, &attr_o);
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_storage.c]: I would prefer
```
res = tee_obj_get(sess->ctx, tee_svc_uref_to_vaddr(attr), &attr_o);
```
as it's more compact

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by pascal-brand-st-dev** [comment]
Updated

File: core/tee/tee_svc_storage.c [hunk 2]
```diff
@@ -586,8 +586,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 	o->pobj = po;
 
 	if (attr != TEE_HANDLE_NULL) {
-		/* init attributes  if provided */
-		res = tee_obj_get(sess->ctx, attr, &attr_o);
+		res = tee_obj_get(sess->ctx,
+			(uint32_t)(uintptr_t)tee_svc_uref_to_kaddr(attr),
```
Comment (jforissier) [inline core/tee/tee_svc_storage.c]: Use  `tee_svc_uref_to_vaddr` (`v` not `k`) and you don't need to cast :)

**Comment by pascal-brand-st-dev** [comment]
Updated

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Pull request merged**